packages/examples/cvat/exchange-oracle/src/crons/_cron_job.py

```diff
@@ -0,0 +1,71 @@
+import inspect
+import logging
+from collections.abc import Callable
+from functools import wraps
+from typing import NamedTuple
+
+from sqlalchemy.orm import Session
+
+from src.db import SessionLocal
+from src.log import get_logger_name
+
+
+class CronSpec(NamedTuple):
+    manage_session: bool
+    repr: str
+
+
+def _validate_cron_function_signature(fn: Callable[..., None]) -> CronSpec:
+    cron_repr = repr(fn.__name__)
+    parameters = dict(inspect.signature(fn).parameters)
+
+    session_param = parameters.pop("session", None)
+    if session_param is not None and session_param.annotation is not Session:
+        raise TypeError(f"{cron_repr} session argument type must be of type {Session.__qualname__}")
+
+    logger_param = parameters.pop("logger", None)
+    if logger_param is None or logger_param.annotation is not logging.Logger:
```

Comment:
Can it be a logger subclass?

Reply:
No. It wouldn't make sense to allow annotation of subclasses here since `cron_job` supplies `logging.Logger`, not its subclasses.